ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -0,0 +1,210 @@
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Analysis (HasAnalysis (..)) where
+
+import           Control.Monad.Except
+import qualified Data.Aeson as Aeson
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import           Data.Foldable (toList)
+import           GHC.Natural (Natural)
+
+import           Cardano.Binary (unAnnotated)
+import qualified Cardano.Crypto as Crypto
+
+import qualified Cardano.Chain.Block as Chain
+import qualified Cardano.Chain.Genesis as Genesis
+import qualified Cardano.Chain.Update as Update
+import qualified Cardano.Chain.UTxO as Chain
+
+import qualified Shelley.Spec.Ledger.BlockChain as SL
+import qualified Shelley.Spec.Ledger.PParams as SL
+import qualified Shelley.Spec.Ledger.Tx as SL
+
+import           Ouroboros.Consensus.Config
+import           Ouroboros.Consensus.Node.ProtocolInfo
+
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock)
+import qualified Ouroboros.Consensus.Byron.Ledger as Byron
+import           Ouroboros.Consensus.Byron.Node (protocolInfoByron)
+
+import           Ouroboros.Consensus.Shelley.Ledger.Block (ShelleyBlock)
+import qualified Ouroboros.Consensus.Shelley.Ledger.Block as Shelley
+import           Ouroboros.Consensus.Shelley.Node (ShelleyGenesis,
+                     protocolInfoShelley)
+import           Ouroboros.Consensus.Shelley.Protocol.Crypto
+
+import           Ouroboros.Consensus.Cardano.Block (CardanoBlock)
+import qualified Ouroboros.Consensus.Cardano.Block as Cardano
+import           Ouroboros.Consensus.Cardano.Node (HardCodedTransition (..),
+                     protocolInfoCardano)
+
+{-------------------------------------------------------------------------------
+  HasAnalysis
+-------------------------------------------------------------------------------}
+
+class HasAnalysis blk where
+    countTxOutputs   :: blk -> Int
+    blockHeaderSize  :: blk -> Natural
+    blockTxs         :: blk -> [BL.ByteString]
```

Comment:
This is meant as a response to https://github.com/input-output-hk/ouroboros-network/pull/2274/#discussion_r444435926, but for some reason GitHub won't let me respond to that issue directly :angry:

> Eventually, I realized I actually use the transactions for 2 things: see numBlockTxs and blockTxsSize. That's why I return the transactions instead of just the size. (I missed this part when we reviewed). I changed the name though, from `blockTxsSize` to `blockTxs`.

I would change it to:
```haskell
blockTxSizes         :: blk -> [Int]
```
Then `blockTxsSize = sum . blockTxSizes` and `blockTxs = length . blockTxSizes`, no need for those bytestrings.